docs/architecture/dapr-for-net-developers/publish-subscribe-buildingblock.md

```diff
@@ -0,0 +1,398 @@
+
+With Dapr, the infrastructure code is **dramatically simplified**. It doesn't need to distinguish between the different message brokers. Dapr provides this abstraction for you. And if needed, you can easily swap out message brokers or configure multiple message broker components.
+
+### Subscribing to events
+
+The earlier eShopOnContainers app contains *SubscriptionManagers* to handle the subscription implementation for each message broker. Each manager contains complex message broker-specific code for handling subscription events. To receive events, each service has to explicitly register a handler for each event-type.
+
+eShopOnDapr streamlines the plumbing for event subscriptions by using Dapr ASP.NET Core libraries. Each event is handled by an action method in the controller. A `Topic` attribute decorates the action method with the name of the corresponding topic to subscribe to. Here's a code snippet taken from the `PaymentService`:
+
+```csharp
+[Route("api/v1/[controller]")]
+[ApiController]
+public class IntegrationEventController : ControllerBase
+{
+    private const string DAPR_PUBSUB_NAME = "pubsub";
+
+    private readonly IServiceProvider _serviceProvider;
+
+    public IntegrationEventController(IServiceProvider serviceProvider)
+    {
+        _serviceProvider = serviceProvider;
+    }
+
+    [HttpPost("OrderStatusChangedToValidated")]
+    [Topic(DAPR_PUBSUB_NAME, "OrderStatusChangedToValidatedIntegrationEvent")]
+    public async Task OrderStarted(OrderStatusChangedToValidatedIntegrationEvent @event)
+    {
+        var handler = _serviceProvider.GetRequiredService<OrderStatusChangedToValidatedIntegrationEventHandler>();
+        await handler.Handle(@event);
+    }
+}
+```
+
+Note in the `Topic` attribute how the name of the event type is used as the topic name to which to subscribe. Then, an event handler that existed in the earlier eShopOnContainers code base is invoked. In the previous example, messages received from the `OrderStatusChangedToValidatedIntegrationEvent` topic invoke the existing  `OrderStatusChangedToValidatedIntegrationEventHandler` event-handler. Because Dapr implements the underlying plumbing for subscriptions and message brokers, a large amount of original code became obsolete and was removed from the code-base. Much of this code was complex to understand and challenging to maintain.
```

Comment:
```suggestion
The `Topic` attribute name corresponds to the event type in the subscription. Then, an event handler that existed in the earlier eShopOnContainers code base is invoked. In the previous example, messages received from the `OrderStatusChangedToValidatedIntegrationEvent` topic invoke the existing  `OrderStatusChangedToValidatedIntegrationEventHandler` event-handler. Because Dapr implements the underlying plumbing for subscriptions and message brokers, a large amount of original code became obsolete and was removed from the code-base. Much of this code was complex to understand and challenging to maintain.
```

Please try to avoid starting sentences with "Note ...".

Reply:
I've changed the first sentence, but not entirely according to your suggestion:

> "In the `Topic` attribute, the name of the .NET type of the event is used as the topic name."

Do you agree with my change?

Reply:
That makes sense to me, thanks.